project_forecast_line/models/project_project.py

```diff
@@ -9,10 +9,24 @@ class ProjectProject(models.Model):
     def _update_forecast_lines_trigger_fields(self):
         return ["stage_id"]
 
+    def _get_written_computed_fields(self, values):
+        # get written and computed field names
+        task_computed = []
+        written_fields = list(values.keys())
+        tree = self.pool.field_triggers.get(self._fields[next(iter(written_fields))])
+        if tree:
+            tocompute = (
+                self.sudo().with_context(active_test=False)._modified_triggers(tree)
+            )
+            for field, _records, _create in tocompute:
+                if field.model_name == "project.project":
+                    task_computed.append(field.name)
+        return written_fields + task_computed
```

Comment:
This whole implementation looks complex to me, and I'm concerned of the speed impact it may have as you recompute this on all create and write calls, whereas the list of values that can trigger a write on another field is fixed. 

I suggest changing the `values` parameter to a sorted tuple of field names (you would call `self._get_written_computed_fields(tuple(sorted(values)))` ) so you can used a cached decorator to prevent recomputing this all the time.